### PR TITLE
Port Modal Header & Styles

### PR DESCRIPTION
**Pull request by zahnno**
Generic Modal header
&
App Styles which gets implemented to ApplicationController.

### CODE REVIEW

File: src/foam/gui/AppStyles.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
+
+foam.CLASS({
+  package: 'foam.gui',
```
Comment (kgrgreer) [inline src/foam/gui/AppStyles.js]: Why the 'gui' package instead of 'u2', where all of the other FOAM UI code is?

File: src/foam/gui/AppStyles.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  name: 'AppStyles',
+  extends: 'foam.u2.View',
+
+  documentation: 'Generic CSS that can be included into the top level controller of foam app. Implement to foam class to use.',
+
+  css: `
```
Comment (kgrgreer) [inline src/foam/gui/AppStyles.js]: Can the DetailView and TableView CSS just be moved to the Detail and TableViews? Can the rest just be moved to the standard Controller?

Comment (zahnno) [reply src/foam/gui/AppStyles.js]: There's no detailview css here. But in regards to the TableView css, it'll overwrite all table view styles in other projects that use the tableview if we append it to the class.

File: src/foam/gui/AppStyles.js (hunk 1 above)
Comment (zahnno) [reply src/foam/gui/AppStyles.js]: it's up to you.

Comment (zahnno) [reply src/foam/gui/AppStyles.js]: I figured these views are a little more detailed comparing to the others in u2

File: src/foam/gui/AppStyles.js (hunk 2 above)
Comment (kgrgreer) [reply src/foam/gui/AppStyles.js]: The following look like they're for DetailView:

 .foam-u2-DetailView {
+      background: #fafafa;
+      border: 1px solid grey;
+    }
+    .foam-u2-DetailView .foam-u2-DetailView {
+      width: auto;
+      margin: inherit;
+    }
+    .foam-u2-DetailView-title {
+      background: #ddd;
+      color: gray;
+      padding: 6px;
+    }

File: src/foam/gui/ModalHeader.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    this.SUPER();
+    var self = this;
+    
+    this
+    .addClass(this.myClass())
+      .start()
+        .start()
+          .start().addClass('container')
+            .start().addClass('title').add(this.title).end()
+            .start({class:'foam.u2.tag.Image', data: '../../gui/images/ic-cancelwhite.svg'}).addClass('close')
```
Comment (kgrgreer) [inline src/foam/gui/ModalHeader.js]: Isn't it better to just set the 'icon' for the action that embed it in an image like this?

File: src/foam/nanos/controller/ApplicationController.js [hunk 4]
```diff
@@ -194,7 +195,7 @@ foam.CLASS({
             s.detach();
           });
         }
-
+
```
Comment (kgrgreer) [inline src/foam/nanos/controller/ApplicationController.js]: Remove trailing whitespace.

File: src/foam/gui/AppStyles.js (hunk 1 above)
Comment (kgrgreer) [reply src/foam/gui/AppStyles.js]: I would move to u2 except for the parts related to to controller, which should move to the same package in nanos where the controller is.

File: src/foam/gui/AppStyles.js (hunk 2 above)
Comment (zahnno) [reply src/foam/gui/AppStyles.js]: sheesh, that slipped my vision.

File: src/foam/gui/ModalHeader.js (hunk 3 above)
Comment (zahnno) [reply src/foam/gui/ModalHeader.js]: Icon isn't showing it

File: src/files.js [hunk 5]
```diff
@@ -144,6 +144,9 @@ FOAM_FILES([
   { name: "foam/web/DetachedURLState", flags: ['web'] },
   { name: "foam/web/URLState", flags: ['web'] },
   { name: "lib/input", flags: ['web'] },
+  { name: 'foam/gui/ModalHeader', flags: ['web'] },
```
Comment (kgrgreer) [inline src/files.js]: Wrong package. Was moved from gui to u2.

File: src/foam/u2/ModalHeader.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
+
+foam.CLASS({
+  package: 'foam.u2',
+  name: 'ModalHeader',
+  extends: 'foam.u2.View',
+
+  documentation: 'Modal Container close/title',
```
Comment (kgrgreer) [inline src/foam/u2/ModalHeader.js]: Maybe make documentation a little clearer.

File: src/foam/u2/ModalHeader.js [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+  imports: [
+    'stack',
+    'closeDialog'
+  ],
+
+  exports: [
```
Comment (kgrgreer) [inline src/foam/u2/ModalHeader.js]: Why do you need to export closeDialog if you're also importing it?

File: src/foam/nanos/controller/AppStyles.js [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+    }
+    tbody > tr > th > td {
+      font-size: 12px;
+      letter-spacing: 0.2px;
+      text-align: left;
+      color: #093649;
+      padding-left: 15px;
+      height: 60px;
+    }
+    .foam-u2-view-TableView th {
```
Comment (kgrgreer) [inline src/foam/nanos/controller/AppStyles.js:43]: Can you move the TableView CSS to the TableView?

**Comment by kennyqiyenkan** [comment]
please note that modal header has been changed on our end.

**Comment by zahnno** [comment]
We'll move to use this modal header once it gets merged. I'll make the changes.

File: src/foam/nanos/controller/AppStyles.js (hunk 8 above)
Comment (zahnno) [reply src/foam/nanos/controller/AppStyles.js:43]: If we append it to the class, It'll overwrite all table view styles in other projects that use the tableview. We'd have to do something similar to what we did with actionviews.
